pandas/tseries/index.py

```diff
@@ -1637,6 +1633,34 @@ def delete(self, loc):
             new_dates = tslib.tz_convert(new_dates, 'UTC', self.tz)
         return DatetimeIndex(new_dates, name=self.name, freq=freq, tz=self.tz)
 
+    def _assert_can_do_op(self, other):
+        zzone = tslib.get_timezone(self.tzinfo)
+        # vzone sholdn't be None if value is non-datetime like
+        vzone = tslib.get_timezone(getattr(other, 'tzinfo', '__no_tz__'))
+        if zzone == vzone:
+            if com.is_list_like(other):
+                return DatetimeIndex(other)
+            else:
+                return Timestamp(other)
+
+        raise ValueError('Passed item and index have different timezone')
+
+    def putmask(self, mask, value):
+        """
+        return a new Index of the values set with the mask
+
+        See also
+        --------
+        numpy.ndarray.putmask
+        """
+        try:
```

Comment:
you don't need this any longer once you call ``_assert_can_do_op`` in ``Index.putmask``